### DISCUSSION TITLE
How to support custom member type?

### DISCUSSION THREAD

**Discussion by aj3423** [pelletier/go-toml#672] [Questions] [answered]
A struct contains `big.Int`, the code doesn't seem to work:
```golang
type Config struct {
	BigInt big.Int
	// lots of other members
}

func main() {
	cfg := &Config{
		BigInt: *big.NewInt(123),
	}

	str, _ := toml.Marshal(cfg)
	fmt.Println(string(str))
}
```

I see there is `MarshalTOML()` interface, but I'll have to implement marshal/unmarshal for all the other fields too, which are too many. Is there any way to support custom field like `big.Int`?

**Comment by pelletier** [comment, marked as answer]
This looks like a bug: I would expect the marshaler to use `big.Int`'s `TextMarshaler()` method. Which version of go-toml are you using?

**Reply by aj3423** [reply]
I tried both v1 and v2, same problem.
	github.com/pelletier/go-toml v1.9.5-0.20211018005641-352072d51afc
	github.com/pelletier/go-toml/v2 v2.0.0-beta.4

**Reply by pelletier** [reply]
Thanks, created https://github.com/pelletier/go-toml/issues/678 to track the resolution.

**Reply by pelletier** [reply]
Should be fixed in v2 as of https://github.com/pelletier/go-toml/commit/2b3de620e8cbec29bb050462bd074456247cd7b4.

It was because `big.Int` actually implements `TextMarshaler` on `*big.Int`, and the go-toml encoder did not consider it when storing it as a value, as opposed to a pointer. Let me know if it works for you!

**Reply by aj3423** [reply]
@pelletier How to update to the v2 to the latest master?  I tried write `github.com/pelletier/go-toml/v2 master` to "go.mod" then `go mod tidy` shows error: 
```
github.com/pelletier/go-toml/v2: version "v2.0.0-20211018005641-352072d51afc" invalid: go.mod has non-.../v2 module path "github.com/pelletier/go-toml" (and .../v2/go.mod does not exist) at revision 352072d51afc
```

**Reply by pelletier** [reply]
For v2 the full module name is `github.com/pelletier/go-toml/v2`. You can use `go get -u` to update your `go.mod` to a specific commit:

```
$ go get -u github.com/pelletier/go-toml/v2@2b3de620e8cbec29bb050462bd074456247cd7b4
```

After this command, your `go.mod` should be automatically updated to contain a line like:

```
github.com/pelletier/go-toml/v2 v2.0.0-beta.4.0.20211124194349-2b3de620e8cb
```
